src/librustc/hir/lowering.rs

```diff
@@ -1149,13 +1149,69 @@ impl<'a> LoweringContext<'a> {
         }
     }
 
-    fn lower_generic_arg(&mut self,
-                         arg: &ast::GenericArg,
-                         itctx: ImplTraitContext<'_>)
-                         -> hir::GenericArg {
+    fn lower_generic_arg(
+        &mut self,
+        arg: &ast::GenericArg,
+        itctx: ImplTraitContext<'_>
+    ) -> hir::GenericArg {
         match arg {
             ast::GenericArg::Lifetime(lt) => GenericArg::Lifetime(self.lower_lifetime(&lt)),
-            ast::GenericArg::Type(ty) => GenericArg::Type(self.lower_ty_direct(&ty, itctx)),
+            ast::GenericArg::Type(ty) => {
+                // A path type argument could be an incorrectly parsed generic const argument. If
```

Comment:
I think it's better to explain this as: "We parse const arguments as type arguments, because we cannot distinguish between them during parsing. We then try to resolve the ambiguity here." (or similar). Saying it's "incorrectly parsed" makes it sound like the compiler's doing something erroneous.